src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -119,10 +119,13 @@ else if (!savedSenderID.equals(senderID)) {
                         editor.commit();
                     }
 
-                    if (gCachedExtras != null) {
+                    if (!gCachedExtras.isEmpty()) {
                         Log.v(LOG_TAG, "sending cached extras");
-                        sendExtras(gCachedExtras);
-                        gCachedExtras = null;
+                        Iterator<Bundle> gCachedExtrasIterator = gCachedExtras.iterator();
+                        while (gCachedExtrasIterator.hasNext()) {
+                            sendExtras(gCachedExtrasIterator.next());
```

Comment:
I have thought of it indeed but decided not to fall into a breaking change for now. I like the idea, are you OK with making that change into the API?